### PR TITLE
feat(ruby): make hit generic

### PR DESCRIPTION
**Pull request by millotp**
## 🧭 What and Why

There is no generic in ruby, so 1 way to make the Hit object accept anything is to add a any getter and setter when serializing/deserializing.

### CODE REVIEW

File: templates/ruby/base_object.mustache [hunk 1]
```diff
@@ -21,6 +21,10 @@
           transformed_hash["#{key}"] = _deserialize(type, attributes[attribute_map[key]])
         end
       end
+      {{#vendorExtensions.x-is-generic}}
+
+      transformed_hash.merge!(attributes.reject { |k, _| attribute_map.key?(k.to_sym) })
```
Comment (shortcuts) [inline templates/ruby/base_object.mustache:26]: that's an horrible syntax ahah but... if it works

**Review by shortcuts — COMMENTED** [review]
looks great but is it expected that it only has effect on the hit model?

**Comment by millotp** [comment]
there should be `getObjectsResponse` too, but maybe it doesn't work because it's an inline model

File: templates/ruby/base_object.mustache (hunk 1 above)
Comment (millotp) [reply templates/ruby/base_object.mustache:26]: Thankfully copilot wrote it for me !

**Comment by shortcuts** [comment]
> there should be `getObjectsResponse` too, but maybe it doesn't work because it's an inline model

is it something we want to fix in this PR?

**Comment by millotp** [comment]
The type is already `Array<Object>` so there is no issue with serialization/deserialization here, it will work as expected

**Review by shortcuts — APPROVED** [review]
gg!

**Pull request merged**